### PR TITLE
Write NetworkManager config files

### PR DESCRIPTION
**Pull request by teclator**
## Problem

During installation, the network configuration written is wicked specific, when the backend selected to be used by the target system is NetworkManager the configuration is lost.

- https://trello.com/c/EdjoT1NN/2256-8-write-networkmanager-configuration-during-installation

## Solution

- Write a basic NetworkManager configuration from the connection configs that are known. (Ethernet, Wireless (open, WPA-PSK), VLAN)

### CODE REVIEW

File: src/lib/y2network/network_manager/connection_config_writer.rb [hunk 1]
```diff
@@ -31,16 +31,24 @@ class ConnectionConfigWriter
       def write(conn, old_conn = nil)
         return if conn == old_conn
 
-        file = CFA::NmConnection.new(SYSTEM_CONNECTIONS_PATH.join(conn.name).sub_ext(FILE_EXT))
+        path = SYSTEM_CONNECTIONS_PATH.join(conn.name).sub_ext(FILE_EXT)
+        file = CFA::NmConnection.new(path)
         handler_class = find_handler_class(conn.type)
         return nil if handler_class.nil?
 
+        ensure_permissions(path) unless ::File.exist?(path)
```
Comment (imobachgs) [inline src/lib/y2network/network_manager/connection_config_writer.rb:39]: You could also pass the mode to the `CFA::NmConnection` constructor if you prefer or even in the `save` method. Anyway, it is OK here.

**Review by imobachgs — COMMENTED** [review]
In general, it looks good as a first version. My main concern at this point is that we are overwriting the whole file (we are not calling `load`). I think we should not be that aggressive, but we can address that in a different PR.

File: src/lib/y2network/network_manager/connection_config_writer.rb (hunk 1 above)
Comment (teclator) [reply src/lib/y2network/network_manager/connection_config_writer.rb:39]: Yep, was not sure about the best place by now, so did it explicitly here by now.

**Comment by imobachgs** [comment]
> In general, it looks good as a first version. My main concern at this point is that we are overwriting the whole file (we are not calling `load`). I think we should not be that aggressive, but we can address that in a different PR.

Anyway, it does not affect the installer use-case, so we can postpone it.

**Review by imobachgs — APPROVED** [review]

File: src/lib/cfa/nm_connection.rb [hunk 2]
```diff
@@ -22,28 +22,68 @@
...
+
     # Constructor
     #
     # @param path [String] File path
     # @param file_handler [.read, .write] Object to read/write the file.
     def initialize(path, file_handler: nil)
-      super(AugeasParser.new("Networkmanager.lns"), path, file_handler: file_handler)
+      @path = path
+      # FIXME: The Networkmanager lense writes the values surrounded by double
+      # quotes which is not valid
```
Comment (mchf) [inline src/lib/cfa/nm_connection.rb:45]: was it reported as a bug to the augeas maintainer?

Comment (teclator) [reply src/lib/cfa/nm_connection.rb:45]: Not yet, is it in my TODO list. Do you want to take care? that would be nice

**Pull request merged**